### PR TITLE
feat: forgot-password revamp

### PR DESCRIPTION
**Pull request by helloiambguedes**
## Summary

New forgot password experience

## Notion link

[](https://www.notion.so/talentprotocol/Revamp-Forgot-Password-Experience-e33c3960f90f400495ffc3d8ca352b96?pvs=4)

## How to test?

This must be tested in dev as we cannot fully do it locally.

## Notes

Let's sync with the virtual wallet release as this may clog dev for some hours if something fails

### CODE REVIEW

File: app/views/onboard/reset_password.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<%= react_component("SetPasswordPage",
+    props: {
+        token: params[:token],
```
Comment (RubenSousaDinis) [inline app/views/onboard/reset_password.erb:3]: This is an example of something we should avoid going forward if we want to decouple the BE and FE. Also, the token and user_id are coming from the URL, right? We can access them in the client by fetching them from the page URL?

File: app/views/onboard/forgot_password.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<%= react_component("ForgotPasswordPage",
```
Comment (RubenSousaDinis) [inline app/views/onboard/forgot_password.erb:1]: File name is missing `.html`

File: app/views/onboard/reset_password.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<%= react_component("SetPasswordPage",
```
Comment (RubenSousaDinis) [inline app/views/onboard/reset_password.erb:1]: File name is missing `.html`

**Review by RubenSousaDinis — COMMENTED** [review]
Added some comments

File: app/views/onboard/reset_password.erb (hunk 1 above)
Comment (helloiambguedes) [reply app/views/onboard/reset_password.erb:3]: We do as I didn't add the url cleaning logic.
Will change.

**Review by RubenSousaDinis — APPROVED** [review]

**Pull request merged**